rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -161,8 +166,20 @@ rmw_create_client(
     _register_type(participant, info->response_type_support_);
   }
 
+  // If FASTRTPS_DEFAULT_PROFILES_FILE defined, fill subscriber attributes with a subscriber profile
+  // located based of topic name defined by _create_topic_name(). If no profile is found, a search
+  // with profile_name "client" is attempted. Else, use the default attributes.
+  std::string topic_name_fallback = "client";
   eprosima::fastrtps::SubscriberAttributes subscriberParam;
-  eprosima::fastrtps::PublisherAttributes publisherParam;
+  eprosima::fastrtps::fixed_string<255> sub_topic_name = _create_topic_name(
+    qos_policies, ros_service_response_prefix, service_name, "Reply");
+  Domain::getDefaultSubscriberAttributes(subscriberParam);
+
+  if (XMLProfileManager::fillSubscriberAttributes(
+      sub_topic_name.to_string(), subscriberParam, false) != XMLP_ret::XML_OK)
```

Comment:
is there a specific return code for "profile not found"?
If there is, it would be good to check in all calls to `fillSubscriberAttributes()` that the return code is either "ok" or "profile not found" and not a different error.

Reply:
`fill*Attributes()` consists in a find in a map. The method returns `XMLP_ret::XML_OK` if found, and `XMLP_ret::XML_ERROR` if not. In that sense, `XMLP_ret::XML_ERROR` is the same as profile not found

Reply:
sounds good